### DISCUSSION TITLE
List of known JupyterLite deployments

### DISCUSSION THREAD

**Discussion by jtpio** [jupyterlite/jupyterlite#827] [Show and tell]
Opening a thread to list known JupyterLite deployments out there.

The goal is to be able to keep track of some of them, and also serve as inspiration for anyone wanting to deploy their own. This could eventually be moved to the documentation.

| Project | Live version | Source |
|-----------|------------------|----------|
| NumPy | https://numpy.org | https://github.com/numpy/numpy.org/blob/8f1c679836de56101e6db56e0dcd3f9bcf7b6207/layouts/partials/shell.html#L18-L22 |
| SymPy | https://www.sympy.org/en/shell.html | https://github.com/sympy/live |
| Pandas | https://pandas.pydata.org/getting_started.html | https://github.com/pandas-dev/pandas/tree/2ef8d147c51afa9dfb5d184c2407227cf9e661db/web/interactive_terminal |
| PyMC | https://www.pymc.io/welcome.html#interactive-demo | |
| Try Jupyter | https://jupyter.org/try-jupyter | https://github.com/jupyter/try-jupyter |
| ipywidgets | https://ipywidgets.readthedocs.io/en/stable/#try-it-online | https://github.com/jupyter-widgets/ipywidgets/blob/3148ca3cb93f9ed8832b3e594279391f0ea9ec94/docs/source/jupyter_lite_config.json |
| xarray | https://xarray.dev |  |
| pyhf | https://pyhf.readthedocs.io/en/stable/#try-out-now-with-jupyterlite | https://github.com/scikit-hep/pyhf/blob/c80f6bec19853b5e2eb0db4b8fca34483a18fba9/docs/generate_jupyterlite_iframe.py |
| Robot Kernel | https://robotkernel.readthedocs.io/en/latest/_/lab/| |
| Astro Python | https://www.astropython.com/astro-lite/lab/ | |
| Demos (games, solvers) | https://ruoyu0088.github.io/jupyterlite_demos/lab/index.html | https://github.com/ruoyu0088/jupyterlite_demos |
| PySport | https://playground.pysport.org/lab/index.html | |
| Galpy | https://www.galpy.org/ | |
| ipycanvas | https://ipycanvas.readthedocs.io/en/latest/lite/lab/ | |
| Sphinx Gallery | https://sphinx-gallery.github.io/stable/index.html | |
| Meetup Python Grenoble | https://meetup-python-grenoble.github.io/lab/ | https://github.com/meetup-python-grenoble/meetup-python-grenoble.github.io |
| LabNbook | https://labnbook.fr/les-labdocs-code-sont-disponibles-dans-labnbook-integration-de-jupyterlite/ | |
| Grenoble INP-UGA | https://phelma-sicom.gricad-pages.univ-grenoble-alpes.fr/1a/3pmpols6-cours/TD_1.html | |

**Comment by bollwyvl** [comment]
Very cool, thanks for starting. 

I'd perhaps clean up the empty args on the robot one.

>  moved to the documentation.

Galleries are tricky, at best... it's probably worth building up a one-project-per YAML/TOML/github issue format that's easy to drive-by pr if we go to docs, as long markdown rows are no fun to write. It would be interesting to know a few more things:
- what does it generally _do_
- what are some _things of interest_: e.g. unique kernels, third-party integration, etc.
- how they are deployed (e.g. RTD, etc.)
- screenshots (automating this is a mugs game for rando sites, though)

Knowing the path to the app root, rather than a specific URL, we could potentially automate some of the following, as much of that data is discoverable "at rest"... though adding the `SHA256SUMS` to baseline build would make that a slam-dunk.

- what version of lite they are using
- what extensions they use
- what kinds of contents they are sharing

Some other from the wild:

- Itkwidgets is 3d data visualizations
  - https://itkwidgets.readthedocs.io/en/latest/_static/retro/notebooks/?path=Hello3DWorld.ipynb

Some i've set up:
- Ipydrawio is a drawing program
  - https://ipydrawio.readthedocs.io/en/stable/demo/index.html
- jupyter-starters has settings-based templates that work in lite
  - https://jupyterstarters.readthedocs.io/en/latest/demo/index.html?starter=todo
  - (though the underlying version is not released yet)
- jupyter-videochat has settings-based rooms that work in lite
  - https://jupyter-videochat.readthedocs.io/en/latest/_static/lite/lab/index.html
- pidgy is a markdown kernel
  - https://pidgy.readthedocs.io/en/latest/_/lab/index.html?path=tangle_examples.ipynb&room=   

I really want to get a solid one working with [formulas](https://formulas.readthedocs.io). Grudgingly, 1-for-1 Excel syntax to drive numpy would be a _fly_ mainstream demo.

**Comment by jtpio** [comment]
Thanks @bollwyvl for sharing this :+1: 

Feel free to update the top comment to add the extra examples.

Agree it would be great to automate it, but it sounds quite complicated since there are many ways of deploying a lite site. The main motivation was to keep at list of known deployments just to have an idea, without guaranteeing they will still be available or working in the future.

**Comment by ew-meetup** [comment]
It's used internally at @meetup as a sort of WYSIWYG documentation for some of our repositories.

JupyterLite (Retro) runs in GitHub Pages for our [nbdev](github.com/fastai/nbdev) repositories. It is used as an alternative to the Quatro documentation that nbdev uses under the hood.
It's a little easier to just maintain the notebook as the raw documentation instead of putting Quatro directives all over the place.

The only thing I've been missing is having the table of contents extension available.

**Reply by jtpio** [reply]
FYI @ew-meetup JupyterLite 0.2.0 (and newer) should now include the table of contents.

**Reply by ew-meetup** [reply]
Very cool. 

We didn't end up using it much by the way. But it's a great project!
